### DISCUSSION TITLE
Analyse the code and generate the diagram

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#69] [Ideas]
This is not about Nodezator, but an idea for a new program. It detects all functions, classes, variables used in a project, and generates a diagram that shows the relations among them. In Nodezator, nodes are created to form the diagram; in that imagined program, functions are analysed to form the diagram.

**Comment by KennedyRichard** [comment]
That's an interesting idea, no doubt. I think such an app would be even more useful if it could take such produced data and also display a kind of report/dashboard showing the main flow of data on the app.

I think the programming language used plays an important role in the viability of this app, that is, it has to be one that allows a lot of introspection, no? Python may have what it takes though, but it would sure be a lot of work. I've heard of some apps that offer such functionality before, but I still don't need such functionality on my current workflow, so I'm don't plan on doing any research on the topic myself for the time being, but it is something that is in the back of my mind.

Thank you for yet another idea, Mr. GrimPixel. Now that the new Nodezator version is released, I can finally get back to answering people on discussions. Thank you for your patience. I'll be doing so between my tasks here and there throughout the week (except Sundays, since it is day of the week where I rest/take care of myself).

I look forward to a lot of interesting discussions.

Peace

**Comment by GrimPixel** [comment]
Thank you for your suppurt of the idea.
I think it can be an extension for VSCodium or VSCode. It can be simply like [Foam](https://foambubble.github.io/), with function names replacing wikilinks.
